src/boundaries_open.c

```diff
@@ -44,7 +44,7 @@ int nghosty = 0;
 int nghostz = 0;
 
 void boundaries_check(void){
-	for (int i=0;i<N;i++){
+	for (int i=N-1;i>=0;i--){ // run through loop backwards so we don't have to recheck same index after removing
```

Comment:
This *could* (I don't know) be significantly slower. I guess it's a tradeoff depending on how many particles escape per step on average in most cases (even with many particles). I would think that number is small, so would probably favour to go through the loop the normal way.